src/main.lua

```diff
@@ -29,19 +29,95 @@ sjson = mods['SGG_Modding-SJSON']
 ---@module 'SGG_Modding-ModUtil'
 modutil = mods['SGG_Modding-ModUtil']
 
----@module 'SGG_Modding-Chalk'
-chalk = mods["SGG_Modding-Chalk"]
 ---@module 'SGG_Modding-ReLoad'
 reload = mods['SGG_Modding-ReLoad']
 
----@module 'config'
-config = chalk.auto 'config.lua'
--- ^ this updates our `.cfg` file in the config folder!
-public.config = config -- so other mods can access our config
+local components_path = rom.path.combine(_PLUGIN.plugins_mod_folder_path,'Components')
+
+component_class_data = setmetatable({},{__mode = 'k'})
+component_instance_data = setmetatable({},{__mode = 'k'})
+
+local component_class_meta = {
+	__call = function(s,...)
+		return s.Create(...)
+	end,
+	__newindex = function(s)
+		error('component classes should not be modified',2)
+	end
+}
+
+public.Components = {}
+
+local per_plugin_key_index = {}
+
+local function generate_guid_key(env)
+	local guid = env._PLUGIN.guid
```

Comment:
Nit: maybe rename this to env_guid or calling_plugin_guid for clarity.  Otherwise this line seems a bit nonsensical out of context:
```lua
local guidKey = _PLUGIN.guid .. '|' .. guid .. '|' ..  tostring(keyi)
```